snuba/querylog/query_metadata.py

```diff
@@ -15,6 +22,107 @@ class QueryStatus(Enum):
     TIMEOUT = "timeout"
 
 
+QUERY_ERROR_MAPPINGS = {
+    ErrorCodes.TOO_SLOW: QueryStatus.TIMEOUT,
+    ErrorCodes.TIMEOUT_EXCEEDED: QueryStatus.TIMEOUT,
+    ErrorCodes.SOCKET_TIMEOUT: QueryStatus.TIMEOUT,
+    ErrorCodes.NETWORK_ERROR: QueryStatus.TIMEOUT,
+}
+
+
+def get_query_status_from_error_codes(code: ErrorCodes) -> QueryStatus | None:
+    return QUERY_ERROR_MAPPINGS.get(code)
```

Comment:
Wouldn't this cause an exception if you pass in a Clickhouse error code which is not mapped. Should there be a default?

Reply:
No, `.get` will return None by default.